src/library.js

```diff
@@ -1625,6 +1723,84 @@ const createPilingJs = (rootElement, initOptions = {}) => {
     resetPileBorder();
   };
 
+  // eslint-disable-next-line consistent-return
+  const expandProperty = objective => {
+    if (objective.constructor === Function) {
+      return objective;
+    }
+    if (objective.constructor === String) {
+      return item => item[objective];
+    }
+  };
+
+  const expandArrangementObjective = arrangementObjective => {
+    if (!Array.isArray(arrangementObjective)) {
+      // eslint-disable-next-line no-param-reassign
+      arrangementObjective = [arrangementObjective];
+    }
+    const expandedArrangementObjective = [];
+
+    arrangementObjective.forEach(objective => {
+      const expandedObjective = {};
+
+      if (objective.constructor !== Object) {
+        expandedObjective.property = expandProperty(objective);
+        expandedObjective.aggregator = values =>
+          values.reduce((average, value) => average + value / values.length, 0);
+        expandedObjective.scale = scaleLinear;
+        expandedObjective.inverse = false;
+      } else {
+        expandedObjective.property = expandProperty(objective.property);
+
+        switch (objective.aggregator) {
+          case 'max':
+            expandedObjective.aggregator = values => Math.max(...values);
+            break;
+          case 'min':
+            expandedObjective.aggregator = values => Math.min(...values);
```

Comment:
Can you turn this into a utility function called `min` to `utils.js`? I am sure we might use it elsewhere.

```javascript
export const min = array => array.reduce((min, v) => v < min ? v : min, Infinity);
```

I've tested both your version and the reduce-based version and somehow the reduce-based version is 50% faster.